Add andino website

JesusSilvaUtrera opened the pull request:
This PR is to add the general structure for the Andino website, not all the content of the pages is added. I will open it as a draft PR first so I can receive some feedback first. Some considerations:

- I added some TODO for the future work.
- The part of the Github Action is added but I can't test it and I don't know if it is well implemented (I am a total newbie with CI/CD), so any feedback there is appreciated.
- ReadTheDocs build is not working right now, I have to take a look why, it said something about permissions but I am admin already in this repo.

CC: @francocipollone @jballoffet

JesusSilvaUtrera commented:
Update: I am having trouble uploading the repo to ReadTheDocs. From waht I have seen, once you have connected your Github account, you should be able to import your repositories without any problem. The repo is found, but it takes the 'main' branch as default, so I wanted to change that in the Settings to take this `add_andino_website` instead as default branch to test if it worked, but as you can see in the image, it is not letting me change it:

![image](https://github.com/Ekumen-OS/andino_docs/assets/79662866/d84c547e-1e76-4ee6-a483-347fa0222b7a)

@francocipollone do you know if this is expected or am I doing something wrong? We can just merge this PR and test it on the main branch directly, but I suppose there is a better solution.

francocipollone commented:
> @francocipollone do you know if this is expected or am I doing something wrong? We can just merge this PR and test it on the main branch directly, but I suppose there is a better solution.

Mjm, it might related to the fact that it doesn't recognize the content of the branch as something that can be uploadable.
If you want just to try that, create a bare minimum content to a separate branch (with some index.html page) and see if it appears in the list of options


btw: Feel free to invite me as admin of the project in readthedocs so maybe I can help you out

JesusSilvaUtrera commented:
> Mjm, it might related to the fact that it doesn't recognize the content of the branch as something that can be uploadable. If you want just to try that, create a bare minimum content to a separate branch (with some index.html page) and see if it appears in the list of options
> 
> btw: Feel free to invite me as admin of the project in readthedocs so maybe I can help you out

I have just invited you to the project. About having another branch with minimum content, it should be the same, because (at least from what I understood) readthedocs looks for the `.readthedocs.yaml` file to configure everything. I will take a look again to the `maliput_documentation` repo and see what I am missing.

JesusSilvaUtrera commented:
@francocipollone all the changes we commented have been addressed, so you can proceed to the revision and tell me if you have thought of anything else. I haven't migrated `andino_hardware` becuase it will be changed when the new assmbly process is finished.

Thanks in advance!

francocipollone commented on README.md:
````diff
@@ -1,2 +1,66 @@
+    ```
+- Install ``sphinx``
+    ```sh
+    pip install sphinx
````
nit: note that you have a requirements.txt file so you can pip install from there directly.

JesusSilvaUtrera replied:
Oops you are totally right I worte that part before creating the `requirements.txt` and I forgot to change it , thanks!

francocipollone reviewed (commented):
Great start for the docs! Thanks for pushing this!
I left only one nit to go

francocipollone reviewed (dismissed):
LGTM!

One extra thing, can we change the website URL to be `andino.readthedocs`?

JesusSilvaUtrera commented:
> One extra thing, can we change the website URL to be `andino.readthedocs`?

@francocipollone Wouldn't we need to create that domain, set the SSL certificate and all of that to achieve this? I tried to add the domain in the Project but it is pending to add the SSL certificate.

I also found this as a workaround, but it is not recommended: https://docs.readthedocs.io/en/stable/faq.html#how-do-i-change-my-project-slug-the-url-your-docs-are-served-at

francocipollone commented:
> > One extra thing, can we change the website URL to be `andino.readthedocs`?
> 
> @francocipollone Wouldn't we need to create that domain, set the SSL certificate and all of that to achieve this? I tried to add the domain in the Project but it is pending to add the SSL certificate.
> 
> I also found this as a workaround, but it is not recommended: https://docs.readthedocs.io/en/stable/faq.html#how-do-i-change-my-project-slug-the-url-your-docs-are-served-at

No need to setup an own domain.

In general it uses the project name + readthedocs.org. 
THe issue is that we used andino_documentation as project name. I renamed this but I think that the URL hasn't changed.
I am ok with deleting the project in readthedocs and create it again using andino as name of the project so we get the andino.readthedocs domain

JesusSilvaUtrera commented:
Aaah okay okay I understood you wrong, I will do that tomorrow when I start the day and let you know how it goes.

JesusSilvaUtrera commented:
@francocipollone I have created the new project and invited you, I deleted the old webhook and added the new one, and the automatic building seems to be working fine:

![image](https://github.com/Ekumen-OS/andino_docs/assets/79662866/bcb5bddf-c3b1-4a30-b83c-10ea54ad4ddc)

However, the videos are not working anymore on the website, but they are still working locally, and I don't get why is this happening, any ideas on that?

francocipollone commented:
> @francocipollone I have created the new project and invited you, I deleted the old webhook and added the new one, and the automatic building seems to be working fine:
> 
> 
> However, the videos are not working anymore on the website, but they are still working locally, and I don't get why is this happening, any ideas on that?

I don't think it is related to the new project being added to the readthedocs because the old page is also broken with respect to the video:
https://andino-documentation.readthedocs.io/en/latest/

JesusSilvaUtrera commented:
> I don't think it is related to the new project being added to the readthedocs because the old page is also broken with respect to the video: https://andino-documentation.readthedocs.io/en/latest/

Mmm true, I would swear that they were working last time I checked them, I don't know what happened, all the photos seem fine, maybe it's just an issue with rtd and using videos, I will check it.

JesusSilvaUtrera commented:
@francocipollone I have managed to fix the videos. Apparently, you can't make the videos work both locally and on RTD, because you need to specify the path differently in each case, I added that in the general README for the future, but now it should be ready to merge, PTAL, thanks!

francocipollone reviewed (approved):
Let's merge it and continue working on top.
The basic setup is achieved which is to any new commit we will have an update in readthedocs.

The content is good so far, probably we will go back to the architecture.
So far so good! Thanks for pushing this!

LGTM
